src/sinks/prometheus/remote_write/config.rs

```diff
@@ -0,0 +1,221 @@
+use http::Uri;
+use snafu::prelude::*;
+
+use crate::{
+    http::HttpClient,
+    sinks::{
+        prelude::*,
+        prometheus::PrometheusRemoteWriteAuth,
+        util::{auth::Auth, http::http_response_retry_logic},
+        UriParseSnafu,
+    },
+};
+
+use super::{
+    service::{build_request, RemoteWriteService},
+    sink::{PrometheusRemoteWriteDefaultBatchSettings, RemoteWriteSink},
+};
+
+#[cfg(feature = "aws-core")]
+use super::Errors;
+
+/// The batch config for remote write.
+#[configurable_component]
+#[derive(Clone, Copy, Debug, Default)]
+pub struct RemoteWriteBatchConfig {
+    #[configurable(derived)]
+    #[serde(flatten)]
+    pub batch_settings: BatchConfig<PrometheusRemoteWriteDefaultBatchSettings>,
+
+    /// Set to true if incremental metrics within the batch should be aggregated.
```

Comment:
Aggregation would apply to both incremental and absolute metrics right? The reason being that we normalize everything to absolute metrics before aggregation.
```suggestion
    /// Whether or not to aggregate metrics within a batch.
```

Reply:
Ah yes. I was misunderstanding things initially..